### DISCUSSION TITLE
how to install istio

### DISCUSSION THREAD

**Discussion by divramod** [vitobotta/hetzner-k3s#216] [General]
hey, 

thx for the awesome cluster setup helper!

i am trying to install istio in a successfull setup cluster.
are there any instructions on how to do that?

after installing istio with istioctl install the loadbalancer is in pending state.
can anyone give me a hint on how to start?

it would be also interesting on how to get it running than with the cert-manager.

i was able to get the cert-manager running with the nginx-setup explained in the example.
now, i would like to switch to istio.
or is it maybe a combination between istio and nginx i need?

<img width="1459" alt="Screenshot 2023-05-14 at 09 46 35" src="https://github.com/vitobotta/hetzner-k3s/assets/1331662/4a6a216f-311b-4dd8-a8aa-25d701c29b3a">

**Comment by divramod** [comment]
i found something, that maybe can serve as a starting point: https://docs.nginx.com/nginx-ingress-controller/tutorials/nginx-ingress-istio/

@vitobotta do you have any idea on how i could start reading about or implementing it?

if successful i would be looking forward to document the approach, so that others could use it to?

**Comment by vitobotta** [comment]
Hi @divramod , unfortunately I haven't touched Istio in ages as I use Linkerd as service mesh for some things. I only use Istio for knative for Kubeflow, but indirectly. Afraid I can't help much with it right now :(

**Comment by nmehlei** [comment]
Hey @divramod,
looks like you may be missing the annotations for the load balancer (see: https://github.com/vitobotta/hetzner-k3s/blob/main/docs/Load_balancers.md). Without these, the load balancer on Hetzner side is not being allocated and therefore the IP remains empty.
